### PR TITLE
Ensure QuarkusJsonbComponentInstanceCreator can be used in unit tests

### LINKED ISSUE #4090: Why Quarkus-JsonB is using ArC?
As I was writing a unit test (not `@QuarkusTest`) for a class that uses `Jsonb.fromJson(String,Class)`, I got a NullPointerException on an adapted type field:

```
    @NotNull
    @JsonbProperty("instant")
    @JsonbTypeAdapter(EpochSecondsAdapter.class)
    private Instant instant;  //UTC epoch second Long
```

I'm not sure that the adapted field is actually what causing the problem but It failed on that particular field so I'm writing it down. The way I solved it is by using `Arc.initialize()` as a `BeforeAll` step in the test and `Arc.shutdown()` as a `AfterAll` step.

Why is JsonB using ArC, and is initializing ArC a good practice or I can do something else to make the test pass?

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #4090

### CODE REVIEW

File: extensions/jsonb/runtime/src/main/java/io/quarkus/jsonb/QuarkusJsonbComponentInstanceCreator.java [hunk 1]
```diff
@@ -34,17 +35,26 @@ public void close() throws IOException {
...
-            }
+            return fallbackCreate(componentClass);
         });
     }
 
+    private <T> Object fallbackCreate(Class<T> componentClass) {
+        try {
+            return componentClass.newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new IllegalStateException("Cannot instantiate Jsonb component: " + componentClass, e);
```
Comment (gsmet) [inline extensions/jsonb/runtime/src/main/java/io/quarkus/jsonb/QuarkusJsonbComponentInstanceCreator.java]:
```suggestion
            throw new IllegalStateException("Cannot instantiate JSON-B component: " + componentClass, e);
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks good to me except for a very minor suggestion.

File: extensions/jsonb/runtime/src/main/java/io/quarkus/jsonb/QuarkusJsonbComponentInstanceCreator.java (hunk 1 above)
Comment (geoand) [reply extensions/jsonb/runtime/src/main/java/io/quarkus/jsonb/QuarkusJsonbComponentInstanceCreator.java]: Fixed :)

**Review by gsmet — APPROVED** [review]

**Comment by mkouba** [comment]
I know it's too late but it looks good ;-).

**Pull request merged**